api/v3/Job/ProcessRecurring.php

```diff
@@ -11,7 +11,7 @@
 function civicrm_api3_job_process_recurring($params) {
   $omnipayProcessors = civicrm_api3('PaymentProcessor', 'get', array('class_name' => 'Payment_OmnipayMultiProcessor'));
   $recurringPayments = civicrm_api3('ContributionRecur', 'get', array(
-    'next_sched_contribution_date' => 'today',
+    'next_sched_contribution_date' => ['BETWEEN' => ['today', 'tomorrow']],
```

Comment:
@eileenmcnaughton What is the expected retrieval of `$recurringPayments` here?

On one of our sites, it seems this line triggers the repeat transaction of payment that was scheduled to run on the next day from today.

For eg. If `next_sched_contribution_date` of a recur payment is `2019-06-04 00:00:00`(4th June) and the job is executed on 3rd June, the above code gets this payment and executes repeattransaction for the contribution. The payment is deducted from the payment processor but for some reason the `next_sched_contribution_date` is not incremented when the completetransaction is called (maybe because it does not match today's date 3rd June?).

As `next_sched_contribution_date` remains as it is, the payment is repeated until either of the below is true 

- the status of the recurring is set to completed, i.e, no of installments are over OR
- if the installment is unlimited, it deducts the payment until the job is executed on next day, i.e, on 1st job execution on 4th June, the `next_sched_contribution_date` is correctly incremented to next date based on frequency interval.

We see 24 repeated payments on a contact which was scheduled to run on 2nd June, out of which 23 payments were recorded on 1st and a single payment is from 2nd which also updated `next_sched_contribution_date` to 2nd July.

Reply:
Can we just update the between operator to something like -

    ['BETWEEN' => [date('Y-m-d 00:00:00'), date('Y-m-d 12:59:59')]]

Reply:
@jitendrapurohit if that works I'm good with it - seems like a timezone inconsistency issue?

Reply:
Have submitted a PR - https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/pull/121 after testing the api call on local.